tests/std/tests/P0355R7_calendars_and_time_zones_io/duration_output_tests.h

```diff
@@ -0,0 +1,111 @@
+        assert(test_duration(duration<int, deci>{42}, "42ds"));
+        assert(test_duration(duration<int, ratio<1>>{42}, "42s"));
+        assert(test_duration(duration<int, deca>{42}, "42das"));
+        assert(test_duration(duration<int, hecto>{42}, "42hs"));
+        assert(test_duration(duration<int, kilo>{42}, "42ks"));
+        assert(test_duration(duration<int, mega>{42}, "42Ms"));
+        assert(test_duration(duration<int, giga>{42}, "42Gs"));
+        assert(test_duration(duration<int, tera>{42}, "42Ts"));
+        assert(test_duration(duration<int, peta>{42}, "42Ps"));
+        assert(test_duration(duration<int, exa>{42}, "42Es"));
+        assert(test_duration(duration<int, ratio<60>>{42}, "42min"));
+        assert(test_duration(duration<int, ratio<60 * 60>>{42}, "42h"));
+        assert(test_duration(duration<int, ratio<60 * 60 * 24>>{42}, "42d"));
+
+        assert(test_duration(duration<int, ratio<2>>{24}, "24[2]s"));
+        assert(test_duration(duration<int, ratio<1, 2>>{24}, "24[1/2]s"));
+        assert(test_duration(duration<int, ratio<22, 7>>{24}, "24[22/7]s"));
+
+        assert(test_duration(duration<float>{0.140625}, "0.140625s"));
+        assert(test_duration_stream_manip<char>());
+
+        assert(test_duration(duration<int, atto>{1}, L"1as"));
+        assert(test_duration(duration<int, femto>{2}, L"2fs"));
+        assert(test_duration(duration<int, pico>{3}, L"3ps"));
+        assert(test_duration(duration<int, nano>{42}, L"42ns"));
+        assert(test_duration(duration<int, micro>{42}, L"42\u00b5\u0073"));
+        assert(test_duration(duration<int, milli>{42}, "42ms"));
+        assert(test_duration(duration<int, centi>{42}, L"42cs"));
+        assert(test_duration(duration<int, deci>{42}, L"42ds"));
+        assert(test_duration(duration<int, ratio<1>>{42}, L"42s"));
+        assert(test_duration(duration<int, deca>{42}, L"42das"));
+        assert(test_duration(duration<int, hecto>{42}, L"42hs"));
+        assert(test_duration(duration<int, kilo>{42}, L"42ks"));
+        assert(test_duration(duration<int, mega>{42}, L"42Ms"));
+        assert(test_duration(duration<int, giga>{42}, L"42Gs"));
+        assert(test_duration(duration<int, tera>{42}, L"42Ts"));
+        assert(test_duration(duration<int, peta>{42}, L"42Ps"));
+        assert(test_duration(duration<int, exa>{42}, L"42Es"));
+        assert(test_duration(duration<int, ratio<60>>{42}, L"42min"));
+        assert(test_duration(duration<int, ratio<60 * 60>>{42}, L"42h"));
+        assert(test_duration(duration<int, ratio<60 * 60 * 24>>{42}, L"42d"));
+
+        assert(test_duration(duration<int, ratio<2>>{24}, L"24[2]s"));
+        assert(test_duration(duration<int, ratio<1, 2>>{24}, L"24[1/2]s"));
+        assert(test_duration(duration<int, ratio<22, 7>>{24}, L"24[22/7]s"));
+
+        assert(test_duration(duration<float>{0.140625}, L"0.140625s"));
+        assert(test_duration_stream_manip<wchar_t>());

```

Comment:
It would be great if we could also add at least one test with a large number of digits to exercise using up most of the buffer in the "general" case (not a specific named literal type).